cfme/infrastructure/virtual_machines.py

```diff
@@ -1666,10 +1690,26 @@ def step(self, *args, **kwargs):
 
 @navigator.register(InfraTemplateCollection, 'TemplatesOnly')
 class TemplatesAll(CFMENavigateStep):
-    VIEW = TemplatesOnlyAllView
     prerequisite = NavigateToSibling('All')
 
+    @property
+    def VIEW(self):  # noqa
+        from cfme.infrastructure.provider import InfraProvider
+        try:
+            if self.obj.parent.one_of(InfraProvider):
+                return ProviderTemplatesView
+        except Exception:
```

Comment:
What's the type of exception that you're trying to handle here? I think this is too much of a diaper.

Reply:
When parent is IPappliance this will cause ` AttributeError: 'IPAppliance' object has no attribute 'one_of'`

Reply:
I should probably just add one_of to IPAppliance ?

Reply:
As per our discussion, I'll use isinstance() and remove the try/except block.

Reply:
I removed usage of one_of and used isinstance() instead.